### PR TITLE
Automatically apply pythonpath

### LINKED ISSUE #1037: PYTHONPATH could be automatically handled
Refactoring `tools.environment_append` context manager to manage `PYTHONPATH` variable.
That way PYTHONPATH would be set automatically before `source`, `build`, `package` and `imports` from self.env_info objects.

### PR DESCRIPTION
**Pull request by lasote**
- [x] Refer to the issue that supports this Pull Request.

- [x] If the issue has missing info, explain the purpose/use case/pain/need that covers this Pull Request.

Some other features claims for better python code reusing. Lately, we kind of decided to promote the "tool" way of reuse python instead of the inheritance, so it reduces the necessary lines of code.

- [x] I've read the [Contributing guide](https://raw.githubusercontent.com/conan-io/conan/develop/.github/CONTRIBUTING.md).
- [x] I've followed the PEP8 style guides for Python code.
- [x] I've opened another PR in the Conan docs repo to the ``develop`` branch, documenting this one. Also adding a description of the changes in the ``changelog.rst`` file. https://github.com/conan-io/docs

#https://github.com/conan-io/docs/pull/479

fixes #1037

### CODE REVIEW

File: conans/model/conan_file.py [hunk 1]
```diff
@@ -77,8 +79,15 @@ def create_exports_sources(conanfile):
         return conanfile.exports_sources
 
 
+@contextmanager
+def _env_and_python(conanfile):
+    with environment_append(conanfile.env):
+        with pythonpath(conanfile):
```
Comment (memsharded) [inline conans/model/conan_file.py]: This might be too slow to be done in all places, by default.

Comment (lasote) [reply conans/model/conan_file.py]: Did you manage to do some performance test?

Comment (memsharded) [reply conans/model/conan_file.py]: 3,5% of extra time.

The context is being applied to ``__config_node()`` in ``DepsBuilder`` which does not apply there, maybe that could be omitted.

Comment (lasote) [reply conans/model/conan_file.py]: Could you repeat the measure now? I excluded the python env from the deps builder

Comment (lasote) [reply conans/model/conan_file.py]: (pending tests)

**Pull request merged**